test-adapter/src/main/scala/org/scalajs/sbttestadapter/ScalaJSRunner.scala

```diff
@@ -61,28 +63,37 @@ final class ScalaJSRunner private[testadapter] (
     taskInfos.map(ScalaJSTask.fromInfo(this, _)).toArray
   }
 
-  def done(): String = synchronized {
-    ensureNotDone()
-
-    stopSlaves()
-
-    master.send("runnerDone")
-    val summary = ComUtils.receiveResponse(master) {
-      case ("ok", summary) => summary
+  def done(): String = {
+    synchronized {
+      ensureNotDone()
     }
```

Comment:
If there is such an assumption, we have to verify it (and as far as I can see, `ensureNotDone` does so). It is [incorrect (discouraged?) behavior][1] for the test framework to not throw an exception. One of the things that are actually specified in the test framework.

[1]: https://github.com/harrah/test-interface/blob/master/src/main/java/sbt/testing/Runner.java#L70